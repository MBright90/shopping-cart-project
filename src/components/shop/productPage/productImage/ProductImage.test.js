import { render, screen } from '@testing-library/react'
import React from 'react'
import { HashRouter } from 'react-router-dom'

import ProductImage from './index'

describe('ProductImage', () => {
  let props
  beforeEach(() => {
    props = {
      image: 'www.example.com/testImage'
    }
  })

  it('renders to match snapshot', () => {
    const { image } = render(<ProductImage {...props} />, {
      wrapper: ({ children }) => <HashRouter basename="/">{children}</HashRouter>
    })
    expect(image).toMatchSnapshot()
  })

  it('renders using the correct url', () => {
    render(<ProductImage {...props} />, {
      wrapper: ({ children }) => <HashRouter basename="/">{children}</HashRouter>
    })
    const cardImage = screen.getByRole('img')
    expect(cardImage).toHaveStyle(`background-image: url(${props.image})`)
  })
})
