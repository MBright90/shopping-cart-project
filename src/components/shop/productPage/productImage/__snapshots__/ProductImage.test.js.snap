// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage renders to match snapshot 1`] = `undefined`;
